controller/Booklet.php

```diff
@@ -55,49 +57,52 @@ class Booklet extends AbstractBookletController
      * @access public
      * @author CRP Henri Tudor - TAO Team - {@link http://www.tao.lu}
      * @return void
+     * @throws common_ext_ExtensionException
      */
-    public function index()
+    public function index(): void
     {
         $this->defaultData();
         $this->setView('index.tpl');
     }
 
     /**
      * Edit action
-     * @throws \tao_models_classes_MissingRequestParameterException
-     * @throws \tao_models_classes_dataBinding_GenerisFormDataBindingException
+     * @throws core_kernel_persistence_Exception
+     * @throws common_ext_ExtensionException
+     * @throws tao_models_classes_MissingRequestParameterException
+     * @throws tao_models_classes_dataBinding_GenerisFormDataBindingException
      */
-    public function editBooklet()
+    public function editBooklet(): void
     {
         $this->defaultData();
 
-        $clazz           = $this->getCurrentClass();
-        $instance        = $this->getCurrentInstance();
-        $myFormContainer = new EditForm($clazz, $instance);
+        $currentInstance = $this->getCurrentInstance();
 
-        $myForm = $myFormContainer->getForm();
+        $this->setData('isPreviewEnabled', $currentInstance->getLabel() !== 'in progress');
```

Comment:
Nice trick.
Any chance to introduce a dedicated booklet-object property for that instead of relying on a label value?

Reply:
It is not possible to detect that instance in the progress by the object, because task runs asynchronously. I see your idea, but the goal to give user visual feedback as well, which means that if label is "in progress", better to do nothing with it. In this case, we need to deal with two properties, that seem looks superfluous.

Reply:
But can't I as a user change the label to `In progress` myself? :)

Reply:
You can, but when the task will be finished, the label will be changed to the proper one. 
From UX it would be nice to make links in the tree also disabled, but I don't know yet how to do it.

Reply:
What if I change it back, and keep it like that?
Then the preview will be forever disabled for me, won't it?

Reply:
Yes. It is a reason why am thinking about way to disable entry in the tree too.